### PR TITLE
oci/client: Add internal/fs package for os.Rename

### PR DESCRIPTION
**Pull request by darkowlzz**
`os.Rename()` fails with LinkError when the source and destination are in
different devices.
The following test fails on machines where /tmp is on a different device:
```
    build_test.go:40: 
        Unexpected error:
            <*os.LinkError | 0xc00050f640>: {
                Op: "rename",
                Old: "./testdata/artifact1987351664",
                New: "/tmp/TestBuild3395247411/001/files.tar.gz",
                Err: <syscall.Errno>0x12,
            }
            rename ./testdata/artifact1987351664 /tmp/TestBuild3395247411/001/files.tar.gz: invalid cross-device link
        occurred
--- FAIL: TestBuild (0.00s)
```

internal/fs is copied from source-controller internal/fs. It has an
implementation of Rename with fallback to handle this failure.

This also fixes the build tests failure which failed on some platforms.

This fixes OCI build failure in the CLI.
Before:
```
$ flux build artifact --path ./docs/ --output /tmp/foo.tgz
► building artifact from ./docs/
✗ bulding artifact failed, error: rename ./docs/298647428 /tmp/foo.tgz: invalid cross-device link
```
With this fix:
```
$ flux build artifact --path ./docs/ --output /tmp/foo.tgz
► building artifact from ./docs/
✔ artifact created at /tmp/foo.tgz
```

### CODE REVIEW

File: oci/client/internal/fs/fs_test.go [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+		}
+	}
+}
+
+func cleanUpDir(dir string) {
+	if runtime.GOOS == "windows" {
+		mu.Lock()
+		exec.Command(`taskkill`, `/F`, `/IM`, `git.exe`).Run()
+		mu.Unlock()
+	}
```
Comment (pjbgf) [inline oci/client/internal/fs/fs_test.go:586]: We can remove this as it adds no value to what we are doing withint he project.

Comment (darkowlzz) [reply oci/client/internal/fs/fs_test.go]: Thanks. Removed it.

**Review by pjbgf — APPROVED** [review]
LGTM

thanks @darkowlzz :bow:

**Review by stefanprodan — APPROVED** [review]
LGTM 

Thanks @darkowlzz

**Pull request merged**